File: single-language-analysis/dataset-files/java/TestSmell123.java
import org.junit.Test;
import static org.junit.Assert.*;

public class TestSmell123 {

    @Test
    public void testPinMustBeOpenedBeforeUseAndIsUnusableAfterBeingClosed() {
        Pin pin = Pins.pin(0);
        
        try {
            pin.value();
            fail("Expected IOError");
        } catch (IOException e) {
            // Expected exception
        }
        
        pin.open();
        try {
            pin.value();
        } finally {
            pin.close();
        }
        
        try {
            pin.value();
            fail("Expected IOError");
        } catch (IOException e) {
            // Expected exception
        }
    }
}

File: single-language-analysis/dataset-files/python/test_smell_123.py
# Original URL: https://github.com/romilly/quick2wire-python-api/blob/c5e21e9d804012efd9d214d18909034b4b898c96/quick2wire/test_gpio.py#L10-L23

class TestGPIO:

    def test_smell_123(self):
        pin = pins.pin(0)
        
        with pytest.raises(IOError):
            pin.value
        
        pin.open()
        try:
            pin.value
        finally:
            pin.close()
        
        with pytest.raises(IOError):
            pin.value